evm_loader/program/src/executor/state.rs

```diff
@@ -123,21 +120,25 @@ impl<'a, B: AccountStorage> ExecutorState<'a, B> {
                 }
             })
             .flatten()
-            .collect::<Vec<_>>();
+            .collect();
+        let metas = into_vector(metas);
 
         if !metas.iter().any(|m| (m.pubkey == address) && m.is_writable) {
-            let account = cache_get_or_insert_account(&self.cache, address, self.backend).await;
+            let account =
+                cache_get_or_insert_account(&self.data.cache, address, self.backend).await;
             return Ok(account);
         }
 
-        let mut accounts = BTreeMap::<Pubkey, OwnedAccountInfo>::new();
+        // Create accounts map on the default heap, as it's not part of the persistent state.
+        let mut accounts = std::collections::BTreeMap::<Pubkey, OwnedAccountInfo>::new();
 
```

Comment:
Just checking, this is `BTreeMap` and not `TreeMap`.

Reply:
It's not part of persistent state, if I correctly understand the logic around. So yes, the standard BTreeMap is suitable here, as we dont want to flood account's persistent heap with unnecessary temporary object.